src/crypto/mod.rs

```diff
@@ -81,17 +80,33 @@ impl EcKeyComponents {
     }
 }
 
-pub trait Crypto: Sized {
-    type RemotePublicKey: RemotePublicKey;
-    type LocalKeyPair: LocalKeyPair;
-    fn generate_ephemeral_keypair() -> Result<Self::LocalKeyPair>;
+pub trait Cryptographer: Send + Sync + 'static {
+    /// Generate a random ephemeral local key pair.
+    fn generate_ephemeral_keypair(&self) -> Result<Box<dyn LocalKeyPair>>;
```

Comment:
See previous comment, moved these methods to make `LocalKeyPair` and `RemotePublicKey` traits object-safe.